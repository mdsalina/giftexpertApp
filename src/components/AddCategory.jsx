import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewValue}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChnace = ({target}) => {
        setInputValue(target.value)   
    }

    //cuando se cargar el formulario
    const OnSumits = (event) => {
        event.preventDefault() //evita que se recargue la pagina
        if(inputValue.trim().length > 2){  //trim elimina espacios en blanco inicio y final
            //onAddCategory(categories=>[inputValue, ...categories])  //con el callback se obtiene el estado anterior y se agrega el nuevo valor
            onNewValue(inputValue)  //se envia el valor al padre
            setInputValue('')
        }
    }


  return (
    <form onSubmit={OnSumits} aria-label="form">
        <input 
        type="text" 
        placeholder="Buscar..."
        value={inputValue}
        onChange={onInputChnace}
        />

    </form>
  )
}

AddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired
}