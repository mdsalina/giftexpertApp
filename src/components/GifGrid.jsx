import { useEffect, useState } from "react"

import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"


export const GifGrid = ({category}) => {

//  const [images, setImages] = useState([])
//
//  const getimages = async () => {
//      const images = await getGifs(category)
//      setImages(images)
//      //console.log(images)
//  }
//  
//
//  useEffect(() => {getimages()}, [category]) // Si la categoría cambia, se vuelve a ejecutar el useEffect, si no cambia, no se ejecuta. Si le hubiera entregado un array vacío, se ejecutaría una sola vez.

//hare lo anterior pero con un custom hook
const {images, isLoading} = useFetchGifs(category)

  return (
    <>
        <h3>{category}</h3>
        {isLoading && <p>Cargando...</p> /*si esta cargando, muestra el mensaje de cargando*/} 
        <div className="card-grid">
            {images.map((img) => (
                <GifItem key={img.id} {...img}/> //...img es igual a title={img.title} url={img.url}, esparse todas sus propiedades
            ))}
        </div>
    </>
  )
}
