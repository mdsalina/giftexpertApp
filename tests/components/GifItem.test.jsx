import { render,screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem test', () => {

  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';

  test('debe de hacer mach con el snapshot', () => {
    const {container}=render(<GifItem title={title} url={url}/>)
    expect(container).toMatchSnapshot();
  })

  test('debe de mostrar la imagen con el URL y el Alt de los props', () => {
    //const {getByAltText} = render(<GifItem title={title} url={url}/>)
    //const img = getByAltText(title);
    //expect(img.src).toBe(url);

    //Otra forma de hacerlo
    //render(<GifItem title={title} url={url}/>)
    //expect(screen.getByRole('img').src).toBe(url);
    //expect(screen.getByRole('img').alt).toBe(title);

    // otra forma de hacerlo
    render(<GifItem title={title} url={url}/>)
    const {src,alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  })

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url}/>)
    expect(screen.getByText(title)).toBeTruthy()
  })
});
