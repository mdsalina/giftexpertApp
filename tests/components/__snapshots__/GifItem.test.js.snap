// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test debe de hacer mach con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
